runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -1263,7 +1263,11 @@ TR::Register *J9::X86::TreeEvaluator::multianewArrayEvaluator(TR::Node *node, TR
    // First Dim zero, only allocate 1 zero-length object array
    int32_t zeroArraySize = TR::Compiler->om.discontiguousArrayHeaderSizeInBytes();
    generateRegMemInstruction(LRegMem(), node, targetReg, generateX86MemoryReference(vmThreadReg, offsetof(J9VMThread, heapAlloc), cg), cg);
-   generateRegMemInstruction(LEARegMem(), node, temp1Reg, generateX86MemoryReference(targetReg, zeroArraySize, cg), cg);
+
+   // Take into account alignment requirements for the size of the zero-length array header
+   int32_t zeroArraySizeAligned = OMR::align(zeroArraySize, TR::Compiler->om.objectAlignmentInBytes());
```

Comment:
It seems like `zeroArraySize` is no longer used anywhere else except when we define `zeroArraySizeAligned` here. Can we just use `TR::Compiler->om.discontiguousArrayHeaderSizeInBytes()` and eliminate `zeroArraySize`? Similarly for Z.

Reply:
Revised in commit e98874bf5b1be4188e76f129a472c4a0434cd525